openedx/core/djangolib/blockstore_cache.py

```diff
@@ -101,6 +101,10 @@ def clear(self):
         cache.delete(cache_key)
 
```

Comment:
I just noticed a major bug above on line 84 🤦 : within `BundleCache`, when we call `cache.set(full_key, value)` we need to also pass in `timeout=None`, because otherwise it will default to expiring after 5 minutes, which defaults the whole purpose of having a versioned cache. We are managing invalidation via changing the key, not expiring the values, so these entries should never expire. This may in fact produce the biggest improvement on the real site compared to all these other optimizations.

Reply:
Whoa!  Excellent catch!  This is going to be the best overall improvement, indeed!